inference-engine/src/transformations/src/transformations/common_optimizations/convert_to_unsigned_nms_gather.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/convert_to_unsigned_nms_gather.hpp"
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/rt_info.hpp>
+#include <queue>
+#include <memory>
+#include "itt.hpp"
+
+using namespace ngraph;
+using namespace std;
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertToUnsignedNmsGather, "ConvertToUnsignedNmsGather", 0);
+
+void convert_gather_indices_to_unsigned(const ngraph::NodeVector& gather_nodes);
+
+ngraph::NodeVector get_NmsGather_destinations(const std::shared_ptr<ngraph::Node>& nms_node);
+
+bool ngraph::pass::ConvertToUnsignedNmsGather::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    MATCHER_SCOPE(MarkNMSGather);
+    bool res = false;
+    for (const auto& op : f->get_ordered_ops()) {
+        // without dynamic_cast since not all NMS versions are inherited from the same base nms
+        if (strcmp(op->get_type_name(), opset8::NonMaxSuppression::type_info.name) != 0)
```

Comment:
I think we can use RTTI, but current approach is also applicable if we also check the version of operation.

Reply:
i didn't check op version intentionally since it should work for all versions